soroban-env-host/src/host/metered_clone.rs

```diff
@@ -43,3 +43,21 @@ impl<K, V> MeteredClone for MeteredOrdMap<K, V> {
         Ok(self.clone())
     }
 }
+
+impl MeteredClone for ScContractCode {
+    fn metered_clone(&self, budget: &Budget) -> Result<Self, HostError> {
+        if let ScContractCode::Wasm(c) = self {
+            budget.charge(CostType::BytesClone, c.len() as u64)?;
+        }
+        Ok(self.clone())
+    }
+}
+
+impl MeteredClone for AccountId {
+    fn metered_clone(&self, budget: &Budget) -> Result<Self, HostError> {
+        match self.0 {
+            PublicKey::PublicKeyTypeEd25519(_) => budget.charge(CostType::BytesClone, 32 as u64)?,
```

Comment:
Do we need to measure heap allocated vs not clones differently? At the moment we're bundling them into the same `BytesClone`, but it is likely to have some different cost.

The AccountId is a fixed size object that doesn't require any heap allocation, so the cost of using it is likely different to the cost of cloning the bytes of things like the wasm contract code.

cc @graydon 

(I say this with limited knowledge about how these things measure up in Rust. My experience with how stack allocated vs heap allocated data structures perform mostly comes from Go, which is likely to be very different. So I acknowledge I might be way off with this message and maybe they are the same cost.)

Reply:
Hmm interesting point. I'm not sure about performance difference in heap vs non-heap clones in Rust, I'll defer that to Graydon. 
Regarding bundling them into the same type, it overestimates the cost by assuming the size is not fixed, which aligns with our goal so it should be ok.